src/module.c

```diff
@@ -1074,213 +1072,535 @@ int RM_CreateSubcommand(RedisModuleCommand *parent, const char *name, RedisModul
+                cmd->key_specs[j].begin_search_type = KSPEC_BS_KEYWORD;
+                cmd->key_specs[j].bs.keyword.keyword = zstrdup(info->key_specs[j].bs.keyword.keyword);
+                cmd->key_specs[j].bs.keyword.startfrom = info->key_specs[j].bs.keyword.startfrom;
+                break;
+            default:
+                serverAssert(0);
+            }
 
-int moduleSetCommandKeySpecBeginSearch(RedisModuleCommand *command, int index, keySpec *spec) {
-    struct redisCommand *cmd = command->rediscmd;
+            switch (info->key_specs[j].find_keys_type) {
+            case 0:
+                /* Omitted. This is shorthand to say that it's a single key. */
+                cmd->key_specs[j].find_keys_type = KSPEC_FK_RANGE;
+                cmd->key_specs[j].fk.range.lastkey = 0;
+                cmd->key_specs[j].fk.range.keystep = 1;
+                cmd->key_specs[j].fk.range.limit = 0;
+                break;
+            case REDISMODULE_KSPEC_FK_UNKNOWN:
+                cmd->key_specs[j].find_keys_type = KSPEC_FK_UNKNOWN;
+                break;
+            case REDISMODULE_KSPEC_FK_RANGE:
+                cmd->key_specs[j].find_keys_type = KSPEC_FK_RANGE;
+                cmd->key_specs[j].fk.range.lastkey = info->key_specs[j].fk.range.lastkey;
+                cmd->key_specs[j].fk.range.keystep = info->key_specs[j].fk.range.keystep;
+                cmd->key_specs[j].fk.range.limit = info->key_specs[j].fk.range.limit;
+                break;
+            case REDISMODULE_KSPEC_FK_KEYNUM:
+                cmd->key_specs[j].find_keys_type = KSPEC_FK_KEYNUM;
+                cmd->key_specs[j].fk.keynum.keynumidx = info->key_specs[j].fk.keynum.keynumidx;
+                cmd->key_specs[j].fk.keynum.firstkey = info->key_specs[j].fk.keynum.firstkey;
+                cmd->key_specs[j].fk.keynum.keystep = info->key_specs[j].fk.keynum.keystep;
+                break;
+            default:
+                serverAssert(0);
+            }
+        }
+        populateCommandLegacyRangeSpec(cmd);
+        populateCommandMovableKeys(cmd);
+    }
 
-    if (index >= cmd->key_specs_num)
-        return REDISMODULE_ERR;
+    if (info->args) {
+        cmd->args = moduleCopyCommandArgs(info->args);
+        /* Populate arg.num_args with the number of subargs, recursively */
+        cmd->num_args = populateArgsStructure(cmd->args);
+    }
 
-    cmd->key_specs[index].begin_search_type = spec->begin_search_type;
-    cmd->key_specs[index].bs = spec->bs;
+    /* Fields added in future versions to be added here, under conditions like
+     * `if (info->version >= 2) { access version 2 fields here }` */
 
     return REDISMODULE_OK;
 }
 
-int moduleSetCommandKeySpecFindKeys(RedisModuleCommand *command, int index, keySpec *spec) {
-    struct redisCommand *cmd = command->rediscmd;
+/* Returns 1 if the command info is valid and 0 otherwise. */
+static int moduleValidateCommandInfo(const RedisModuleCommandInfo *info) {
 
```

Comment:
redundant empty line

Reply:
Oh, I added it intentionally. I think the following comment and code are better separated from the start of the function... But I can delete it the empty line if you prefer.